### ISSUE TITLE
Changes in DNS resolution are not backwards compatible

### ISSUE THREAD

**Issue by sallner** [flyingcircusio/batou#189]
I generally like the changes https://github.com/flyingcircusio/batou/commit/a60de7449e576cbf3c39e82b12a9cdb489d86cae which fixes issue https://github.com/flyingcircusio/batou/issues/158 and introduced an explicit declaration of available IPv4 or IPv6 addresses, but as it is implemented now, it is not (in all cases) backwards compatible. I had a deployment with an nginx configuration in the form of 

```
    listen {{component.http.listen}};
    {% if component.http.listen_v6 %}
    listen {{component.http.listen_v6}};
    {% endif %}
```
The current default arguments to `Address` never resolve to an IPv6 address, so this would be an empty line. Interestingly, I found the error, as I had two different sections for a Let's Encrypt ACME-Challenge and one was contacted by IPv4 and the other by IPv6. The latter one failed, as there was no configuration left. This kind of errors could be spotted by a change in the configuration file via diff during batou run, but when deploying to a fresh state or VM, this behaviour change would be unnoticed. 

I wanted to ask, what are the arguments against setting also `resolvev6=True` as default. In case the deployment is not able to resolve to a IPv6 address, that would need an intervention, but I do not know, how much deployments would be affected here. 

Alternatively it should be highlighted more in the changelog with this case in mind or the version could be bumped to 3.0.

**Comment by ctheune** [comment]
This is on purpose. Highlighting sounds good to me.

The old implementation was a bug: we experienced multiple times where buggy DNS servers would sometimes not resolve IPv6 and thus your setup would suddenly break. Even worse: it might work during predictions but then not work during actual deployments.

In this scenario if you expect IPv6 and the resolver does not deliver it, it will explicitly break. If you want V6 to be conditional you now need to handle this explicitly in the deployment.

**Comment by ctheune** [comment]
Alright, I agree that we should make upgrades less prone to accidents as I want this to remain a minor change.

Suggestion: extend the util.Adress class so that any attribute access to `connect`, `listen` and `listen_v6` will cause an error (see batou configuration error classes) if the value is None.

This suppresses implicit accidents where users opportunistically used v6 if available but then are prone to DNS issues that we tried to fix in #158
